store/store.go

```diff
@@ -493,35 +538,29 @@ func (s *SnapUbuntuStoreRepository) Snap(name, channel string, devmode bool, aut
 	case resp.StatusCode == 404:
 		return nil, ErrSnapNotFound
 	case resp.StatusCode != 200:
-		tpl := "Ubuntu CPI service returned unexpected HTTP status code %d while looking for snap %q in channel %q"
+		// TODO: do this for all requests
+		tpl := "Ubuntu CPI service returned unexpected HTTP status code %d while getting details for snap %q in channel %q"
 		if oops := resp.Header.Get("X-Oops-Id"); oops != "" {
 			tpl += " [%s]"
 			return nil, fmt.Errorf(tpl, resp.StatusCode, name, channel, oops)
 		}
 		return nil, fmt.Errorf(tpl, resp.StatusCode, name, channel)
 	}
 
+	s.checkStoreResponse(resp)
```

Comment:
old stuff but this method has a terrible name actually (it's not checking things, it's capturing out-of-band global store prefences, maybe time to find a better one?

also from looking at what it does it should be probably be called only once we are fully sure the response was sane, so closer to  infoFromRemote